apis/core/lib/domain/table/create.ts

```diff
@@ -0,0 +1,34 @@
+import { GraphPointer } from 'clownface'
+import { NamedNode } from 'rdf-js'
+import { schema, rdf } from '@tpluscode/rdf-ns-builders'
+import { cc, shape } from '@cube-creator/core/namespace'
+import { ResourceStore } from '../../ResourceStore'
+import * as id from '../identifiers'
+import { resourceStore } from '../resources'
+
+interface CreateTableCommand {
+  tableCollection: NamedNode
+  resource: GraphPointer
+  store?: ResourceStore
+}
+
+export async function createTable({
+  tableCollection,
+  resource,
+  store = resourceStore(),
+}: CreateTableCommand): Promise<GraphPointer> {
+  const label = resource.out(schema.name).term!
+
+  const table = await store
+    .createMember(tableCollection, id.table(label.value))
+
+  table.addOut(schema.name, label)
+  // TableShape
+  if (shape('table/create').equals(resource.out(cc.Table).term)) {
+    table
+      .addOut(rdf.type, cc.Table)
+  }
```

Comment:
@tpluscode what's the best way to get the triples that were posted to this endpoint and add all of them to the `table` dataset?

Reply:
I tried to build the endpoint and had the exact same question.

Reply:
The request payload is parsed from `await req.dataset()`

To add to the dataset you'd likely loop over that to ensure correct graph URI in the table's dataset

```javascript
[...payloadData].forEach(quad => {
  table.addOut(quad.property, quad.subject)
})
```

That would be the simplest way which assumes that the payload is only triples where the tables is the subject of what you create. Obviously a deeper or more complex graph would require a more specialised approach.